lib/pages/dashboard/widgets/timeline/gymnasium_location.dart

```diff
@@ -0,0 +1,173 @@
+import 'package:add_2_calendar/add_2_calendar.dart' as addToCalendar;
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_map/flutter_map.dart';
+import 'package:latlong/latlong.dart';
+import 'package:map_launcher/map_launcher.dart' as mapLauncher;
+import 'package:v34/commons/loading.dart';
+import 'package:v34/models/event.dart';
+import 'package:v34/pages/dashboard/blocs/gymnasium_bloc.dart';
+import 'package:v34/utils/extensions.dart';
+
+class GymnasiumLocation extends StatefulWidget {
+  final Event event;
+
+  const GymnasiumLocation({Key key, @required this.event}) : super(key: key);
+
+  @override
+  GymnasiumLocationState createState() => GymnasiumLocationState();
+}
+
+class GymnasiumLocationState extends State<GymnasiumLocation> {
+  GymnasiumBloc _gymnasiumBloc;
+
+  @override
+  void initState() {
+    super.initState();
+    _gymnasiumBloc = GymnasiumBloc(RepositoryProvider.of(context), GymnasiumUninitializedState());
+    _gymnasiumBloc.add(LoadGymnasiumEvent(gymnasiumCode: widget.event.gymnasiumCode));
+  }
+
+  void _addEventToCalendar() async {
+    final addToCalendar.Event event = addToCalendar.Event(
+      title: widget.event.name,
+      location: widget.event.place,
+      startDate: widget.event.date,
+      endDate: widget.event.date.add(Duration(hours: 2))
+    );
+    await addToCalendar.Add2Calendar.addEvent2Cal(event);
+  }
+
+  void _launchMap(GymnasiumLoadedState state, bool route) async {
+    try {
+      final coordinates = mapLauncher.Coords(state.gymnasium.latitude, state.gymnasium.longitude);
+      final title = state.gymnasium.name;
+      final availableMaps = await mapLauncher.MapLauncher.installedMaps;
+
+      if (availableMaps.length == 1) {
+        if (route) {
+          availableMaps.first.showDirections(
+            destination: coordinates,
+            destinationTitle: title
+          );
+        } else {
+          availableMaps.first.showMarker(
+            coords: coordinates,
+            title: title
+          );
+        }
+      } else {
+        showModalBottomSheet(
+          context: context,
+          builder: (BuildContext context) {
+            return SafeArea(
+              child: SingleChildScrollView(
+                child: Container(
+                  child: Wrap(
+                    children: <Widget>[
+                      for (var map in availableMaps)
+                        ListTile(
+                          onTap: () {
+                            if (route) {
+                              map.showDirections(
```

Comment:
I don't know why but when I tried this branch on my phone (with 2 maps installed: Waze + Google Maps), even if I click on Waze in the bottom sheet then it always launches Google Maps...

Reply:
Did you pulled the last commit ? It was a bug on the first commit ! I tried the new version with Google Maps and Waze and it well launches Waze ! The fix is exactly at the line you are showing !